addon/components/section.js

```diff
@@ -30,15 +30,16 @@ export default Component.extend(PropTypeMixin, {
    */
   init () {
     this._super()
-    this.set('state', Ember.Object.create({
-      expanded: this.get('expanded') || this.get('expandedOnInitialRender')
-    }))
+    this.set('state', Ember.Object.create())
+    if (this.get('expanded') === undefined) {
+      this.set('expanded', this.get('expandedOnInitialRender'))
+    }
```

Comment:
`expanded` is a property that is to be set by the consumer of this component and thus the component shouldn't be mutating it, that is why it was on the `state` property which was meant to contain internal state.